File: csharp/accumulate/Accumulate.cs
using System;
using System.Collections.Generic;

public static class AccumulateExtensions
{
    public static IEnumerable<TResult> Accumulate<TSource, TResult>(this IEnumerable<TSource> collection, Func<TSource, TResult> map)
    {
        foreach (var element in collection)
            yield return map(element);
    }
}

File: vbnet/accumulate/Accumulate.vb
Imports System.Runtime.CompilerServices

Module AccumulateExtensions
    
    <Extension> 
    Public Iterator Function Accumulate(Of T)(aSource As IEnumerable(Of T), aMap As Func(Of T, T)) As IEnumerable(Of T)
        For Each element In aSource
            Yield aMap(element)
        Next
    End Function

End Module